src/redux/reducers/process-resources.js

```diff
@@ -0,0 +1,97 @@
+// processResource() executes once per/resource, as paginated FHIR requests resolve.
+
+const getType = (resource) => {
+  const { resourceType } = resource;
+  if (resourceType === 'Observation') {
+    const { code } = resource.category[0].coding[0];
+    switch (code) {
+      case 'laboratory':
+      case 'vital-signs':
+        return code;
+      default: {
+        console.info(`Unsupported code type for Observation ${resource.id}: `, code); // eslint-disable-line no-console
+        return 'UNSUPPORTED_OBSERVATION';
+      }
+    }
+  }
+  return resourceType; // could be one of: RESOURCE_TYPES, PATIENT, or something unsupported
+};
+
+const getSubType = (resource) => {
+  let subType;
+  switch (resource.resourceType) {
+    case 'Condition':
+    case 'DiagnosticReport':
+    case 'Procedure':
+    case 'Observation':
+      subType = resource.code?.text;
+      break;
+    case 'Encounter':
+      subType = resource.type?.[0]?.text;
+      break;
+    case 'Immunization':
+      subType = resource.vaccineCode?.text;
+      break;
+    case 'MedicationRequest':
+      subType = resource.medicationCodeableConcept?.text;
+      break;
+    case 'CarePlan':
+      subType = resource.category?.[0]?.text;
+      break;
+    default:
+      console.warn(`No subType found for resource: ${resource}`); // eslint-disable-line no-console
+      subType = 'Other';
+      break;
+  }
+
+  return subType;
+};
+
+const getTimelineDate = (resource) => {
```

Comment:
This one would be used by components. I'm fine having this live here though.

Reply:
components could just use `resource.timelineDate`, directly.